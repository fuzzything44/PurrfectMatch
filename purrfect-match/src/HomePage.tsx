import React from 'react';
import { Redirect } from "react-router-dom";

interface HomePageProps {
};

interface HomePageState {
};

export class HomePage extends React.Component<HomePageProps, HomePageState> {

    constructor(props: any) {
        super(props);

        this.state = {
        };
    }

    public render() {

        return (
            <div>
                Home page.
            </div>
        );
    }

}


