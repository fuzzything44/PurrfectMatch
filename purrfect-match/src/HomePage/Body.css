.body-wrapper {
  width: 90%;
  max-width: 1100px;
  margin-right: auto;
  margin-left: auto;
  text-align: center;
  margin-bottom: 3em;
}