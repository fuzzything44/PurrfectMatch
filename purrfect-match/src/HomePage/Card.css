.Card {
  width: 24vw;
  height: 40vw;
  position: relative;
  overflow: hidden;
  border-radius: 1.5vw;
  box-shadow: 0 1.5vw 3vw rgba(0, 0, 0, 0.25);
  display: grid;
  grid-template-rows: 1fr 1fr;
  transition: 0.8s cubic-bezier(0.2, 0.8, 0.2, 1);
  transform: translateY(0);
}

.Card:hover {
  transform: scale(1.05, 1.05);
  box-shadow: 0 2vw 4vw rgba(0, 0, 0, 0.5);
}

.Card:hover img {
  transform: translateY(-0.2vw);
}

.Card img {
  position: absolute;
  top: 0;
  height: 100%;
  z-index: -1;
  transition: 0.8s cubic-bezier(0.2, 0.8, 0.2, 1);
}

