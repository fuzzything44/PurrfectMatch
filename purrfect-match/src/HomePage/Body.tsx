import React from 'react';
import './Body.css';

interface BodyProps {
};

interface BodyState {
};

export default class Body extends React.Component<BodyProps, BodyState> {
  constructor(props: any) {
    super(props);

    this.state = {
    };
  }

  public render() {
    return (
      <div>
        <div className="body-wrapper">
          <div className="feature-1">
            <h1 className="feature-header">1. Find your perfect pet!</h1>
            <p className="paragraph-header">We've partnered with hundreds of animal shelters to help you with the process.</p>
          </div>
          <div className="feature-2">
            <h1 className="feature-header">2. Contact the shelter.</h1>
            <p className="paragraph-header">Each pet should have contact information for its respective shelter.</p>
          </div>
          <div className="feature-3">
            <h1 className="feature-header">3. Pick up your companion.</h1>
            <p className="paragraph-header">Once you pick up your companion you can leave a success story and be featured on the site.</p>
          </div>
        </div>
      </div >
    )
  }
}

