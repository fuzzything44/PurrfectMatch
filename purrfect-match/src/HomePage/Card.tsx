import React from 'react';
import './Card.css';

interface CardProps {
  image: string;
}

interface CardState {
};

export class Card extends React.Component<CardProps, CardState> {
  constructor(props: any) {
    super(props);

    this.state = {
    };
  }

  public render() {
    return (
      <div className="Card">
        <img src={this.props.image} />
      </div>
    )
  }
}