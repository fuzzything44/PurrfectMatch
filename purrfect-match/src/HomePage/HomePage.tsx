import React from 'react';
import { Redirect } from "react-router-dom";
import { Card } from './Card';


interface HomePageProps {
};

interface HomePageState {
};

export class HomePage extends React.Component<HomePageProps, HomePageState> {
    constructor(props: any) {
        super(props);

        this.state = {
        };
    }

    public render() {
        return (
            <div>
                <div className="Hero">
                    <div className="PetSelectionBox">
                        <div className="Cards">
                            <h2>Find your perfect companion from the comfort of your own home.</h2>
                            <h2>Choose which type of animal you would like to start with:</h2>
                            <div className="CardGroup">
                                <div>
                                    <h1>Dog</h1>
                                    <Card image={require('../pictures/dog-homepage.jpg')} />
                                </div>
                                <div>

                                    <h1>Cat</h1>
                                    <Card image={require('../pictures/cat-homepage.jpg')} />
                                </div>
                                <div>
                                    <h1>Other</h1>
                                    <Card image={require('../pictures/other-animal-homepage.jpg')} />
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
                <div className="heroSpacer" />
            </div>
        );
    }
}


