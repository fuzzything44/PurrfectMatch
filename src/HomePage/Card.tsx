import React from 'react';
import './Card.css';

interface CardProps {
    title : string;
    text? : string;
    image : string;
}

const Card = (props : CardProps) => (
    <div className="Card">
        <img src={props.image} />
        <h3>{props.title}</h3>
        <p>{props.text}</p>
    </div>
)

export default Card