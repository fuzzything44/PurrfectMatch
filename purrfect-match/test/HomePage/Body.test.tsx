import React from 'react';
import ReactDOM from 'react-dom';
import { configure, shallow, ShallowWrapper } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Body } from 'HomePage/Body';

configure({ adapter: new Adapter() });

let wrapper: ShallowWrapper<any, any, any>;
beforeEach(() => {
  wrapper = shallow(<Body />);
});

test("Renders correctly", () => {
  expect(wrapper).toMatchSnapshot();
});
