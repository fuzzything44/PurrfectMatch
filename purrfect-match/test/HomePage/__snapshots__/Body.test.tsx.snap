// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div>
  <div
    className="body-wrapper"
  >
    <div>
      <h1
        className="feature-header"
      >
        1. Find your perfect pet!
      </h1>
      <p
        className="paragraph-header"
      >
        We've partnered with hundreds of animal shelters to help you with the process.
      </p>
    </div>
    <div>
      <h1
        className="feature-header"
      >
        2. Contact the shelter.
      </h1>
      <p
        className="paragraph-header"
      >
        Each pet should have contact information for its respective shelter.
      </p>
    </div>
    <div>
      <h1
        className="feature-header"
      >
        3. Pick up your companion.
      </h1>
      <p
        className="paragraph-header"
      >
        Once you pick up your companion you can leave a success story and be featured on the site.
      </p>
    </div>
  </div>
</div>
`;
